packages/react-components/react-card/e2e/Card.e2e.tsx

```diff
@@ -0,0 +1,243 @@
+          description={<span>Developer</span>}
+        />
+        <div>
+          Donut chocolate bar oat cake. Dragée tiramisu lollipop bear claw. Marshmallow pastry jujubes toffee sugar
+          plum.
+        </div>
+        <CardFooter>
+          <Button id="open-button" onClick={alert} appearance="primary">
+            Open
+          </Button>
+          <Button id="close-button" appearance="outline">
+            Close
+          </Button>
+        </CardFooter>
+      </Card>
+      <p tabIndex={0} id="after">
+        After
+      </p>
+    </>
+  );
+};
+
+describe('Card', () => {
+  describe('focus behaviors', () => {
+    describe('focusMode="off" (default)', () => {
+      it('should not be focusable', () => {
+        mountFluent(<CardSample />);
+
+        cy.get('#before').focus();
+
+        cy.get('#card').should('not.be.focused');
+
+        cy.realPress('Tab');
+
+        cy.get('#card').should('not.be.focused');
+        cy.get('#open-button').should('be.focused');
+      });
+    });
+
+    describe('focusMode="no-tab"', () => {
+      it('should be focusable', () => {
+        mountFluent(<CardSample focusMode="no-tab" />);
+
+        const card = cy.get('#card');
+
+        card.should('not.be.focused');
+
+        card.focus();
+
+        card.should('be.focused');
+      });
+
+      it('should focus inner elements on EnterKey press', () => {
+        mountFluent(<CardSample focusMode="no-tab" />);
+
+        cy.get('#card').focus();
+
+        cy.realPress('Enter');
+
+        cy.get('#open-button').should('be.focused');
+      });
+
+      it('should not focus inner elements on Tab press', () => {
+        mountFluent(<CardSample focusMode="no-tab" />);
+
+        cy.get('#card').focus();
+
+        cy.realPress('Tab');
+
+        cy.get('#card').should('not.be.focused');
+        cy.get('#after').should('be.focused');
+      });
+
+      it('should trap focus', () => {
+        mountFluent(<CardSample focusMode="no-tab" />);
+
+        cy.get('#open-button').focus();
+
+        cy.realPress('Tab');
+
+        cy.get('#open-button').should('not.be.focused');
+        cy.get('#close-button').should('be.focused');
+
+        cy.realPress('Tab');
+
+        cy.get('#open-button').should('be.focused');
+        cy.get('#close-button').should('not.be.focused');
+      });
+
+      it('should focus parent on Esc press', () => {
+        mountFluent(<CardSample focusMode="no-tab" />);
+
+        cy.get('#open-button').focus();
+
+        cy.realPress('Escape');
+
+        cy.get('#open-button').should('not.be.focused');
+        cy.get('#card').should('be.focused');
+      });
+    });
+
+    describe('focusMode="tab-exit"', () => {
+      it('should be focusable', () => {
+        mountFluent(<CardSample focusMode="tab-exit" />);
+
+        const card = cy.get('#card');
+
+        card.should('not.be.focused');
+
+        card.focus();
+
+        card.should('be.focused');
+      });
```

Comment:
nit: the focusable card test could be in another test suite. You could use `forEach` to iterate through different behaviours and only write the test once. You can take a look at `Popover` e2e tests where it goes through controlled and uncontrolled.

Sometimes I also think duping tests can be nice because reusable tests are always a bit less readable. So up to you if you want to do this or not

Reply:
I prefer to go with verbosity over efficiency for tests. Tests should be as readable as possible overall and the less "magic" we add to them, the better.

Reply:
the ancient wisdom says: "it depends" :D